### ISSUE TITLE
clang-format: lines are too long

### ISSUE THREAD

**Issue by scrubbbbs** [nomacs/nomacs#1193]
The indent width is set to 160. This is very unusual and too often cumbersome. I think it should be 90 or 100.

I would like to be able to fit two editors on my screen side-by-side, or one editor and some documentation on the other side, without clipping which is not possible on a 1440p screen, even with toolbars minimized and tiny font. See screenshot below which is at 150cols wide.

![image](https://github.com/user-attachments/assets/8e916f90-24c6-4028-b440-a568703c4310)

**Comment by leejuyuu** [comment]
@scrubbbbs Have you tried setting text wrapping? Most editors should support that. I do not use Qt Creator but this seems like it https://doc.qt.io/qtcreator/creator-indenting-code.html

I would vote for **not** disabling the check. The auto-formatting for long lines is still useful.

120 columns sounds like a good compromise.

**Comment by novomesk** [comment]
Change from 160->120 is going to produce lot of changed files.
I let @scrubbbbs to do it when **Theme: New Light and Dark Themes** is merged.

**Comment by scrubbbbs** [comment]
Yeah the merge conflicts would be a pain...good call. Theme PR is almost ready for (hopefully final) review

**Comment by scrubbbbs** [comment]
As @leejuyuu pointed out, we should probably convert CRLF to LF at this time as well #1296.

And on other lint-related issues we should also do a pass on cmakefiles/python/anything skipped by clang-format, for tabs->spaces

**Comment by scrubbbbs** [comment]
I'm working on a plan to do this, I think this is a logical order

- dos2unix
- code transformations for C++17 mainly to reduce line wrapping. 
  - Replacing default constructed values with braces `{}`. Most of this can be done with regex
  - Other ideas?
- clang-format everything that currently isn't compliant
- modify clang-format file for 120 or anything else we want and reformat
- cmake files: tabs to spaces and maybe auto-format if I can find a tool for it

**Comment by leejuyuu** [comment]
https://github.com/BlankSpruce/gersemi looks promising for cmake format.




>  * code transformations for C++17 mainly to reduce line wrapping.
>       
>    * Replacing default constructed values with braces `{}`. Most of this can be done with regex
>    * Other ideas?

Are there examples of what you want to achieve? Additionally, since this step alters the code semantically, I suggest moving it to the last step. (So that for pure formatting changes, we can skip through reviews)

**Comment by scrubbbbs** [comment]
> Are there examples of what you want to achieve? 

I'm testing some regex to do this now. It won't always prevent wrapping (when column count is reduced), but often helps. I think it is easier to read otherwise.

The first two changes below prevent wrapping (at 120 width). The third one is just for terseness/readability.

```diff
--- ImageLounge/src/DkGui/DkCentralWidget.h
+++ ImageLounge/src/DkGui/DkCentralWidget.h
@@ -81,3 +81,3 @@ public:
 
-    DkTabInfo(const QSharedPointer<DkImageContainerT> imgC = QSharedPointer<DkImageContainerT>(), int idx = -1, QObject *parent = 0);
+    DkTabInfo(const QSharedPointer<DkImageContainerT> imgC = {}, int idx = -1, QObject *parent = 0);
     DkTabInfo(TabMode mode, int idx = -1, QObject *parent = 0);
@@ -157,3 +157,3 @@ public slots:
     void setTabList(QVector<QSharedPointer<DkTabInfo>> tabInfos, int activeIndex = -1);
-    void addTab(QSharedPointer<DkImageContainerT> imgC = QSharedPointer<DkImageContainerT>(), int tabIdx = -1, bool background = false);
+    void addTab(QSharedPointer<DkImageContainerT> imgC = {}, int tabIdx = -1, bool background = false);
     void addTab(const QString &filePath, int idx = -1, bool background = false);
@@ -174,3 +174,3 @@ public slots:
     void loadUrls(const QList<QUrl> &urls, const int maxUrlsToLoad = 20);
-    void openBatch(const QStringList &selectedFiles = QStringList());
+    void openBatch(const QStringList &selectedFiles = {});
     void showBatch(bool show = true);
```

> Additionally, since this step alters the code semantically, I suggest moving it to the last step. (So that for pure formatting changes, we can skip through reviews)

If I understand correctly, the only possible change that could go unnoticed would be if the class also has a constructor with an initializer list, and they do different things. I should probably check the source code of  QString, QStringList, QVector, but I don't think anything in Qt will be an issue.

**Comment by scrubbbbs** [comment]
> https://github.com/BlankSpruce/gersemi looks promising for cmake format.

That seems to work, thanks!

**Comment by scrubbbbs** [comment]
I want to change the wrapping for arguments to functions/initializers. This will reduce number of lines changed substantially (in this example, from +3 lines when wrapping to +1)

Currently, when wrapping they are all placed on one line. I can appreciate the added readability but with 2x the line count for the same block seems a bit much. This mainly seen when using connect() as it takes a lot of arguments and often has to wrap.

This is enabled with the `BinPack*` clang-format settings.

Without bin pack:
```c++
        connect(am.action(DkActionManager::preview_select_all),
                &QAction::triggered,
                mThumbsScene,
                &DkThumbScene::selectAllThumbs);
        connect(am.action(DkActionManager::preview_zoom_in),
                &QAction::triggered,
                mThumbsScene,
                &DkThumbScene::increaseThumbs);
        connect(am.action(DkActionManager::preview_zoom_out),
                &QAction::triggered,
                mThumbsScene,
                &DkThumbScene::decreaseThumbs);
```

With:

```c++
        connect(am.action(DkActionManager::preview_select_all), &QAction::triggered, mThumbsScene,
                &DkThumbScene::selectAllThumbs);
        connect(am.action(DkActionManager::preview_zoom_in), &QAction::triggered, mThumbsScene,
                &DkThumbScene::increaseThumbs);
        connect(am.action(DkActionManager::preview_zoom_out), &QAction::triggered, mThumbsScene,
                &DkThumbScene::decreaseThumbs);
```

Update: I changed my mind on this one in the pull request...separate lines it is! I don't see the line increase as too big of a deal. A lot of the cases where this happens is because of long line syndrome and that code can be refactored when we get to it.

**Comment by leejuyuu** [comment]
I think a larger number of lines won't be an issue.
